File: backend/src/main/java/com/rental/car/Models/UserAuth/RegisterRequest.java
package com.rental.car.Models.UserAuth;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data @Builder
@AllArgsConstructor @NoArgsConstructor
public class RegisterRequest {
    private String firstname;
    private String lastname;
    private String email;
    private String password;
    private String nationality;
    private String identifier;
    private String borndate;
}

File: frontend/src/models/RegisterRequest.ts
export class RegisterRequest{
  public firstname:string;
  public lastname:string;
  public email:string;
  public nationality:string;
  public identifier:number;
  public password:string;
  public borndate:string;

  constructor(
    firstname:string,
    lastname:string,
    email:string,
    nationality:string,
    borndate:string,
    identifier:number,
    password:string
  ){
    this.firstname = firstname;
    this.lastname = lastname;
    this.email = email;
    this.nationality = nationality;
    this.identifier = identifier;
    this.password = password;
    this.borndate = borndate;
  }
}